src/libraries/System.Text.Json/tests/System.Text.Json.Tests/NewtonsoftTests/JsonSerializerTests.cs

```diff
@@ -157,4 +163,14 @@ public class ChildClass : BaseClass
     {
         public new bool IsTransient { get; set; }
     }
+
+    public class Base
```

Comment:
Thanks! Would it be possible to add source generator tests as well?

Reply:
Test rewrritten as a `PropertyVisibility` generic test, run for both Reflection and Source Generator use cases.
Unfortunately, the test fails in the SrcGen case. I would need some help here. The getter cannot be set afterwards.

Reply:
Yeah, that is probably expected. There is a corresponding method in the source generator that should be synced:

https://github.com/dotnet/runtime/blob/3ae99a78d8967e5f19765c52f272febef5def623/src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs#L1000-L1051

Reply:
Method updated accordingly, but does not generate the code correctly. It seems that I modify the derived property too late...

Reply:
>  It seems that I modify the derived property too late...

Can you clarify what you mean by that? An example of a failing test and what the error is would help.